### DISCUSSION TITLE
Coqui VITs spectrogram extraction

### DISCUSSION THREAD

**Discussion by AlyShmahell** [coqui-ai/TTS#2433] [General Q&A] [answered]
Hello,

I was wondering if `notebooks/ExtractTTSpectrogram.ipynb` is suitable for VITs as well as Tacotron, I've been following the code in the notebook in an attempt to extract a spectrogram from a pre-trained VITs model for use in another piece of software I'm writing for audio synthesis (tested to work with ground truth spectrograms I have), but so far the output is ineligible with VITs, any info or suggestions are appreciated, thanks.
Note, for testing VITs, I used the `Synthesizer.tts` function and it does output wav files correctly (for their values being integers I just use a normalization function to get them between -1 and 1), so the pretrained model works, it's just that the notebook instructions on getting a spectrogram instead of audio is what doesn't work correctly.

**Comment by erogol** [comment, marked as answer]
by VITS it is not possible to get melspectrogram output. It uses its own hidden representation instead of spectrogram

**Reply by AlyShmahell** [reply]
Thank you for the reply.

I would appreciate it if you could answer another one, the pipeline for VITs `forward` function is the following:
`z_slice -> waveform_decoder = HifiganGenerator -> model_outputs` isn't `z_slice` in this case some form of spectrogram? or is it still a custom/hidden representation?
